p2p/libp2p/module.go

```diff
@@ -0,0 +1,338 @@
+package libp2p
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"github.com/libp2p/go-libp2p"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/pokt-network/pocket/logger"
+	"github.com/pokt-network/pocket/p2p/common"
+	"github.com/pokt-network/pocket/p2p/libp2p/identity"
+	"github.com/pokt-network/pocket/p2p/libp2p/protocol"
+	"github.com/pokt-network/pocket/p2p/providers"
+	"github.com/pokt-network/pocket/p2p/providers/addrbook_provider"
+	"github.com/pokt-network/pocket/p2p/providers/addrbook_provider/persistence"
+	"github.com/pokt-network/pocket/p2p/stdnetwork"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/runtime/configs"
+	poktCrypto "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+var _ modules.P2PModule = &libp2pModule{}
+
+type libp2pModule struct {
+	log                   *modules.Logger
+	bus                   modules.Bus
+	cfg                   *configs.P2PConfig
+	addrBookProvider      providers.AddrBookProvider
+	currentHeightProvider providers.CurrentHeightProvider
+	identity              libp2p.Option
+	listenAddrs           libp2p.Option
+	host                  host.Host
+	pubsub                *pubsub.PubSub
+	topic                 *pubsub.Topic
+	subscription          *pubsub.Subscription
+	network               typesP2P.Network
+}
+
+// NewReadStreamDeadline returns a future deadline
+// based on the read stream timeout duration.
+func NewReadStreamDeadline() time.Time {
```

Comment:
Doesn't need to be exposed outside of this package - please lowercase.

Ditto other functions